src/mbgl/util/mapbox.cpp

```diff
@@ -35,19 +34,19 @@ std::vector<std::string> getMapboxURLPathname(const std::string& url) {
     return pathname;
 }
 
-std::string normalizeSourceURL(const std::string& url, const std::string& accessToken) {
+std::string normalizeSourceURL(const std::string& baseURL, const std::string& url, const std::string& accessToken) {
     if (!isMapboxURL(url)) {
         return url;
     }
 
     if (accessToken.empty()) {
         throw std::runtime_error("You must provide a Mapbox API access token for Mapbox tile sources");
     }
-
+    
     return baseURL + "v4/" + url.substr(protocol.length()) + ".json?access_token=" + accessToken + "&secure";
```

Comment:
After removing the slash from the default value, add a slash here (`"/v4/"`) and in each of the other normalization functions.